aws/ci_build.rake

```diff
@@ -203,16 +197,18 @@ def upgrade_frontend(name, host)
     'git pull --ff-only',
     'sudo bundle install',
     'rake build',
+    # Use the AWS-provided script to cleanly re-register a frontend instance with its load balancer(s).
+    'bin/deploy_frontend/register_with_elb.sh'
   ]
   command = commands.join(' && ')
 
   HipChat.log "Upgrading <b>#{name}</b> (#{host})..."
 
   log_path = aws_dir "deploy-#{name}.log"
 
-  # Stop the frontend before running the commands so that the git pull doesn't modify files
-  # out from under a running instance. The rake build command will restart the instance.
-  stop_frontend name, host, log_path
+  # Remove the frontend from load balancer rotation before running the commands,
+  # so that the git pull doesn't modify files out from under a running instance.
+  deregister_frontend name, host, log_path
```

Comment:
I think we need to ensure that we detect errors from this call and log to Slack with a serious error so that a human can take action; we might also want to consider adding our own retry logic because it's a bad thing for the deploy to fail in the middle.  Even it requires multiple retries over many minutes, that's still better than the alternative.

As Mehal notes, deregister_frontend can fail, although I assume it has some internal retries that make this not too likely.  If this does happen, I believe this will blow up and we'll be left in a state where only some of the frontends are upgraded.   That's a bad 

Without the warning to slack and retries, it's possible the DoTD won't even realize the deploy has stopped.